ports/stm32/boards/make-pins.py

```diff
@@ -378,9 +379,31 @@ def print_af_hdr(self, af_const_filename):
                       file=af_const_file)
                 print_conditional_endif(cond_var, file=af_const_file)
 
+    def print_af_defs(self, af_defs_filename):
+        with open(af_defs_filename,  'wt') as af_defs_file:
+            STATIC_AF = set()
+            seen = []
+
+            print("enum STATIC_AF_TOKENS {", file=af_defs_file)
+            for named_pin in self.board_pins:
+                for af in named_pin.pin().alt_fn:
+                    func = "%s%d" % (af.func, af.fn_num) if af.fn_num else af.func
+                    pin_type = (af.pin_type or "NULL").split('(')[0]
+                    STATIC_AF.add("    (pin_obj) == (pin_%s) && STATIC_AF_ ## fn ## _ ## type == STATIC_AF_%s_%s ? (%d) : \\" % (
```

Comment:
I'd suggest to change it to just `STATIC_AF_ ## fn_type`, and pass in fn and type as one entity, eg `SDIO_D0`.